src/main/java/com/labsynch/labseer/service/SaltServiceImpl.java

```diff
@@ -246,21 +246,21 @@ public ArrayList<ErrorMessage> validateSaltEdit(Salt oldSalt, Salt newSalt)
 		}
 
 		List<Salt> saltsByName = Salt.findSaltsByNameEquals(newSalt.getName()).getResultList();
-		if (saltsByName.size() > 0) {
+		if (saltsByName.size() > 0 && !newSalt.getName().equals(oldSalt.getName())) {
```

Comment:
nit: I think this will work, but the logic is a little confusing to me. We're doing a search, then ignoring the results if the salt name didn't change.
Could we skip the search entirely if the salt name hasn't changed?

Or the other approach that makes sense to me is to always do the search, then filter out the "current salt" from the results by ID, and see what (if any) other results are left.

Reply:
I like the idea of skipping the search if there hasn't been a change. If the salt was allowed to be registered in the first place with the name and/or abbreviation then there shouldn't be another so the search would theoretically be ineffective.